pkg/storage/aws/s3/storage.go

```diff
@@ -40,7 +47,69 @@ func (s *Storage) Get(ctx context.Context, path string) storage.IResponse {
 	})
 	s3Err := <-responseChannel
 
-	return storage.NewResponse([]byte(buff.Bytes()), getStatusCodeFromError(s3Err), s3Err)
+	return storage.NewResponse(buff.Bytes(), getStatusCodeFromError(s3Err, nil), s3Err)
+}
+
+// GetPartialObject takes in the Context, path and opt as an argument and returns an IResponse interface implementation.
+// This method figures out how to get partial data from the S3 storage backend.
+func (s *Storage) GetPartialObject(ctx context.Context, path string, opt *storage.GetPartialObjectRequestOptions) storage.IResponse {
+	if opt == nil || len(opt.Range) == 0 {
+		return s.Get(ctx, path)
+	}
+
+	input := s3.GetObjectInput{
+		Bucket: aws.String(s.bucketName),
+		Key:    aws.String(path),
+		Range:  &opt.Range,
+	}
+	type getObjectResponse struct {
+		output *s3.GetObjectOutput
+		err    error
+	}
+	responseChannel := make(chan getObjectResponse, 1)
+	makeNetworkCall(s.hystrixCmd.Name, s.hystrixCmd.Config, func() error {
```

Comment:
the hystrix name is still using the global hystrix config (it's not customised according to the method). should we use different hystrix name for each method?

Reply:
@baskarap your suggestion 👆🏼?

Reply:
IMO, we should use a different hystrix command name (and probably under a different configuration) so that we can track this call separately in darkroom's dashboard.

Reply:
do you have a suggestion for the config name (something like: `hystrixGetPartialObjectCmd` maybe? Also, the current configuration called `hystrixCmd` which kinda generic, should keep this as is?

Reply:
Hmm, the generic one was created because supporting video was never in the plan earlier. If we keep it like this, will it be possible to use 2 different circuit breaker configurations? For GetObject and GetPartialObject.

Reply:
We can have `storage.HystrixCommand` as a part of `GetPartialObjectRequestOptions` and that option will override the global `storage.HystrixCommand`. What do you think?